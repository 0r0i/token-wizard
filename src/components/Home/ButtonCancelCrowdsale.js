import React, { Component } from 'react'
import logdown from 'logdown'
import cancelDeploy from '../../utils/cancelDeploy'

const logger = logdown('TW:Home:CancelCrowdsale')

export class ButtonCancelCrowdsale extends Component {
  cancel = () => {
    logger.log('Cancel crowdsale')
    cancelDeploy()
  }

  render() {
    return (
      <button onClick={() => this.cancel()} className="hm-ButtonCancelCrowdsale">
        Cancel crowdsale
      </button>
    )
  }
}
