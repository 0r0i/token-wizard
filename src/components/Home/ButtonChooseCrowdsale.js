import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'

export class ButtonChooseCrowdsale extends Component {
  goToCrowdsales = () => {
    localStorage.reload = true
    navigateTo({
      history: this.props.history,
      location: 'crowdsales',
      fromLocation: 'home'
    })
  }

  render() {
    return (
      <div onClick={() => this.goToCrowdsales()} className="hm-ButtonChooseCrowdsale">
        Choose Crowdsale
      </div>
    )
  }
}
