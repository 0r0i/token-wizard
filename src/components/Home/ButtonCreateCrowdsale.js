import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'

export class ButtonCreateCrowdsale extends Component {
  goToStepOne = () => {
    localStorage.reload = true
    navigateTo({
      history: this.props.history,
      location: 'stepOne',
      fromLocation: 'home'
    })
  }

  render() {
    return (
      <button onClick={() => this.goToStepOne()} className="hm-ButtonCreateCrowdsale">
        New crowdsale
      </button>
    )
  }
}
