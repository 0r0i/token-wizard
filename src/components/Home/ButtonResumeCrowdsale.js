import React, { Component } from 'react'

export class ButtonResumeCrowdsale extends Component {
  resume = () => {
    localStorage.reload = true
    window.location = '/4'
  }

  render() {
    return (
      <button onClick={() => this.resume()} className="hm-ButtonResumeCrowdsale">
        Resume crowdsale
      </button>
    )
  }
}
