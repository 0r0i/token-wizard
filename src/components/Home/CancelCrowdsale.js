import React, { Component } from 'react'
import logdown from 'logdown'
import cancelDeploy from '../../utils/cancelDeploy'

const logger = logdown('TW:Home:CancelCrowdsale')

export class CancelCrowdsale extends Component {
  cancel = () => {
    logger.log('Cancel crowdsale')
    cancelDeploy()
  }

  render() {
    return (
      <button onClick={() => this.cancel()} className="hm-Home_BtnChoose">
        Cancel crowdsale
      </button>
    )
  }
}
