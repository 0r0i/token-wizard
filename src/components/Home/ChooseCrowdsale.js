import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'

export class ChooseCrowdsale extends Component {
  goToCrowdsales = () => {
    navigateTo(this.props, 'crowdsales')
  }

  render() {
    return (
      <div onClick={() => this.goToCrowdsales()} className="hm-Home_BtnChoose">
        Choose Crowdsale
      </div>
    )
  }
}
