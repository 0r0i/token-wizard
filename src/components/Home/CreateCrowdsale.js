import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'
import logdown from 'logdown'

const logger = logdown('TW:Home:CreateCrowdsale')

export class CreateCrowdsale extends Component {
  goToStepOne = () => {
    try {
      localStorage.reload = true
      navigateTo({
        history: this.props.history,
        location: 'stepOne',
        fromLocation: 'home'
      })
    } catch (err) {
      logger.log('Error to navigate', err)
    }
  }

  render() {
    return (
      <button onClick={() => this.goToStepOne()} className="hm-Home_BtnNew">
        New crowdsale
      </button>
    )
  }
}
