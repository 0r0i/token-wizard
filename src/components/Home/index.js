import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import CrowdsalesList from '../Common/CrowdsalesList'
import { Loader } from '../Common/Loader'
import { loadRegistryAddresses } from '../../utils/blockchainHelpers'
import { ModalContainer } from '../Common/ModalContainer'
import { toast } from '../../utils/utils'
import { TOAST, NAVIGATION_STEPS, DOWNLOAD_STATUS } from '../../utils/constants'
import { inject, observer } from 'mobx-react'
import { checkWeb3, getNetworkVersion } from '../../utils/blockchainHelpers'
import { getCrowdsaleAssets } from '../../stores/utils'
import logdown from 'logdown'
import { LogoPrimary } from '../LogoPrimary/index'

const logger = logdown('TW:home')

const { CROWDSALE_STRATEGY, TOKEN_SETUP, CROWDSALE_SETUP, PUBLISH, CROWDSALE_PAGE } = NAVIGATION_STEPS

@inject('web3Store', 'generalStore', 'contractStore')
@observer
export class Home extends Component {
  constructor(props) {
    super(props)
    this.state = {
      showModal: false,
      loading: false
    }

    let { contractStore } = this.props
    contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.PENDING)
  }

  componentDidMount() {
    let { generalStore, web3Store, contractStore } = this.props
    checkWeb3(web3Store.web3)

    getNetworkVersion()
      .then(networkID => {
        generalStore.setProperty('networkID', networkID)
        getCrowdsaleAssets(networkID)
      })
      .then(
        () => {
          contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.SUCCESS)
        },
        e => {
          logger.error('Error downloading contracts', e)
          toast.showToaster({
            type: TOAST.TYPE.ERROR,
            message:
              'The contracts could not be downloaded.Please try to refresh the page. If the problem persists, try again later.'
          })

          contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.FAILURE)
        }
      )
  }

  chooseContract = () => {
    this.setState({
      loading: true
    })

    loadRegistryAddresses().then(
      () => {
        this.setState({
          loading: false,
          showModal: true
        })
      },
      e => {
        logger.error('There was a problem loading the crowdsale addresses from the registry', e)
        this.setState({
          loading: false
        })
      }
    )
  }

  onClick = crowdsaleAddress => {
    this.props.history.push('/manage/' + crowdsaleAddress)
  }

  hideModal = () => {
    this.setState({ showModal: false })
  }

  render() {
    return (
      <div>
        <section className="hm-Home">
          <div className="hm-Home_Scroll">
            <div className="hm-Home_Top">
              <LogoPrimary />
            </div>
            <div className="hm-Home_MainInfoContainer">
              <div className="hm-Home_MainInfo">
                <div className="hm-Home_MainInfoTextContainer">
                  <h1 className="hm-Home_MainInfoTitle">Welcome to Token Wizard</h1>
                  <p className="hm-Home_MainInfoDescription">
                    Token Wizard is a client side tool to create ERC20 token and crowdsale in five steps. It helps you
                    to publish contracts on the Ethereum network, create a crowdsale page with stats. For participants,
                    the wizard creates a page to contribute into the campaign.
                  </p>
                  <p className="hm-Home_MainInfoPoweredBy">
                    Token Wizard is powered by <a href="https://github.com/auth-os/beta">Auth-os</a>.
                  </p>
                </div>
                <div className="hm-Home_MainInfoButtonContainer">
                  <Link className="hm-Home_BtnNew" to="/1">
                    New crowdsale
                  </Link>
                  <div onClick={() => this.chooseContract()} className="hm-Home_BtnChoose">
                    Choose Crowdsale
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        {/* <ModalContainer
          title={'Crowdsale List'}
          description={`The list of your updatable crowdsales. Choose crowdsale address, click Continue and you'll be
        able to update the parameters of crowdsale.`}
          // hideModal={this.hideModal}
          // showModal={this.state.showModal}
          showModal={true}
        >
          <CrowdsalesList onClick={this.onClick} />
        </ModalContainer> */}
        <Loader show={this.state.loading} />
      </div>
    )
  }
}
