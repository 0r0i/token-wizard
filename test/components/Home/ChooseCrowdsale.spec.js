import React from 'react'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow, render } from 'enzyme'
import { MemoryRouter } from 'react-router'
import { ChooseCrowdsale } from '../../../src/components/Home/ChooseCrowdsale'

configure({ adapter: new Adapter() })

describe('ChooseCrowdsale', () => {
  const history = {
    push: jest.fn()
  }

  it('should render screen with shallow without throwing an error', () => {
    const wrapper = shallow(<ChooseCrowdsale />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render screen with mount without throwing an error', () => {
    // Given
    const wrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <ChooseCrowdsale />
      </MemoryRouter>
    )

    // When
    const tree = wrapper.find('.hm-Home_BtnChoose')

    // Then
    expect(tree).toHaveLength(1)
  })

  it('should render screen with render without throwing an error', () => {
    // Given
    const wrapper = render(
      <MemoryRouter initialEntries={['/']}>
        <ChooseCrowdsale />
      </MemoryRouter>
    )
    // When
    const tree = wrapper.html()

    // Then
    expect(tree).toMatchSnapshot()
  })

  it(`should simulate click`, () => {
    // Given
    const wrapper = shallow(<ChooseCrowdsale history={history} />)
    const instance = wrapper.instance()
    const goToCrowdsalesHandler = jest.spyOn(instance, 'goToCrowdsales')
    instance.forceUpdate()
    // When
    const buttonChooseCrowdsale = wrapper.find('.hm-Home_BtnChoose')
    buttonChooseCrowdsale.simulate('click')

    // Then
    expect(buttonChooseCrowdsale.length).toBe(1)
    expect(goToCrowdsalesHandler).toHaveBeenCalledTimes(1)
  })
})
