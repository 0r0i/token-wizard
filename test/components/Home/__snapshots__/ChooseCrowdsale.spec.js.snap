// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseCrowdsale should render screen with render without throwing an error 1`] = `"Choose Crowdsale"`;

exports[`ChooseCrowdsale should render screen with shallow without throwing an error 1`] = `
<div
  className="hm-Home_BtnChoose"
  onClick={[Function]}
>
  Choose Crowdsale
</div>
`;
